src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -3090,6 +3267,57 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
         eval quick_visualization_preview((value) model.nodes.set_quick_preview(*value));
     }
 
+
+
+    // =============
+    // === Modes ===
+    // =============
+
+    let profiling_mode_transition = Animation::new(network);
+    frp::extend! { network
+        out.source.mode <+ frp.toggle_profiling_mode.map2(&frp.mode,|_,&mode| {
+            match mode {
+                Mode::Profiling => Mode::Normal,
+                _               => Mode::Profiling,
```

Comment:
please make FRP functions short. Refactor this as method on `Mode`. Please apply that to all lambdas in FRP network definitions that you can apply to (just make them as short and clear as possible)